pandas/io/formats/excel.py

```diff
@@ -649,20 +649,21 @@ def _format_header_regular(self) -> Iterable[ExcelCell]:
                 )
 
     def _format_header(self) -> Iterable[ExcelCell]:
+        gen: Iterable[ExcelCell]
+
         if isinstance(self.columns, MultiIndex):
             gen = self._format_header_mi()
         else:
             gen = self._format_header_regular()
 
-        gen2 = ()
+        gen2: Iterable[ExcelCell] | tuple[()] = ()
```

Comment:
Is mypy happy when you remove `tuple[()]`? A tuple is(?) also an iterable.

Reply:
@twoertwein mypy is happy when I remove ````tuple[()]```` and leave it just an an Iterable . I was wanting to be explicit with the definition because ````gen2````  would remain an empty tuple  when the if clause doesn't pass. If its not required, Ill fix it

Reply:
Great! I think it would then be better to remove the tuple part. An empty tuple is still an iterable - just with no elements.

Reply:
updated